src/sentry/models/group.py

```diff
@@ -118,6 +118,38 @@ def from_kwargs(self, project, **kwargs):
                 }
             )
 
+    def from_event_id(self, project, event_id):
+        """
+        Resolves the 32 character event_id string into
+        a Group for which it is found.
+        """
+        from sentry.models import EventMapping, Event
+        group_id = None
+
+        # Look up event_id in both Event and EventMapping,
+        # and bail when it matches one of them, prioritizing
+        # Event since it contains more history.
+        for model in Event, EventMapping:
+            try:
+                group_id = model.objects.filter(
+                    project_id=project.id,
+                    event_id=event_id,
+                ).values_list('group_id', flat=True)[0]
+
+                # It's possible that group_id is NULL
```

Comment:
Good catch/thinking here.

Out of curiosity, under what circumstances would this actually happen?

Reply:
I assume historically, the column is nullable. I don't think practically it's a thing to worry about, but tests failed from this.